state/factory/statetx.go

```diff
@@ -178,7 +179,29 @@ func (stx *stateTX) PutState(pkHash hash.Hash160, s interface{}) error {
 		return errors.Wrapf(err, "failed to convert account %v to bytes", s)
 	}
 	stx.cb.Put(AccountKVNameSpace, pkHash[:], ss, "error when putting k = %x", pkHash)
-	return nil
+	return stx.putIndex(pkHash, ss)
+}
+
+func (stx *stateTX) getMaxVersion(pkHash hash.Hash160) (uint64, error) {
```

Comment:
This is the typical way,iterate from max version,this will be used for api query:
err = boltdb.View(func(tx *bolt.Tx) error {
		c := tx.Bucket([]byte(AccountKVNameSpace)).Cursor()
		for k, v := c.Seek(maxStateKey); k != nil; k, v = c.Prev() {
			kHeight := binary.BigEndian.Uint64(k[20:])
			if kHeight <= height {if err := state.Deserialize(s, v); err != nil {
					return errors.Wrapf(err, "error when deserializing state data into %T", s)
				}
				return nil
			}
		}
		return errors.New("cannot find state")
	})